exercises/concept/logs-logs-logs/.docs/instructions.md

```diff
@@ -1,56 +1,116 @@
 
 ```go
-Message("[ERROR]: Invalid operation")
-// Output: "Invalid operation"
+Application("forecast: ☀ sunny")
+// Output: weather
 ```
 
-Any leading or trailing white space should be removed:
+```go
+Application("❗ recommended search product 🔍")
+// Output: recommendation
+```
 
 ```go
-Message("[WARNING]:  Disk almost full\r\n")
-// Output: "Disk almost full"
+Application("search complete")
+// Output: default
+```
+
+## 2. Redact invalid characters
+
+Logs may contain invalid, illegal, or even profane characters. As such, it may be necessary to redact information from the logs before they are stored in a log aggregation system.
+
+Implement the `Redact` function using the following function signature:
+
+```
+func Redact(log string, redactions []rune) string
 ```
 
-## 2. Get the message length in characters
+The `Redact` function should read the `log`, redact all occurances of the characters in `redactions` from `log`, and return the redacted log line to the caller.
 
-Implement the `MessageLen` function to return a log line's message length:
+```go
+log := "please redact '👎' characters: 👎 thank you"
+redactions := []rune{'👎'}
+
+Redact(log, redactions)
+// Output: please redact '' characters:  thank you
+```
 
 ```go
-MessageLen("[ERROR]: Invalid operation \n")
-// Output: 17
+log := "XXplease redact '👎' and 'X' characters: 👎 thank youXX"
+redactions := []rune{'👎', 'X'}
+
+Redact(log, redactions)
+// Output: please redact '' and '' characters:  thank you"
+```
+
+## 3. Replace incorrect characters
+
+Logs may contain incorrect characters. As such, these characters may need to be replaced with other characters before the log can be stored in a log aggregation system.
+
+Implement the `Replace` function using the following function signature:
+
+```
+func Replace(log string, old, new rune) string
 ```
 
-## 3. Get log level from a log line
+The `Replace` function should read the `log`, replace all occurances of the `old` character with the `new` character within `log`, and return the modified log line to the caller.
 
-Implement the `LogLevel` function to return a log line's log level, which should be returned in lowercase:
+```go
+log := "please replace '👎' with '👍'"
+
+Replace(log, '👎', '👍')
+// Output: please replace '👍' with '👍'"
+```
 
 ```go
-LogLevel("[ERROR]: Invalid operation")
-// Output: "error"
+log := "please replace 'e' with 'X'"
+
+Replace(log, 'e', 'X')
+// Output: plXasX rXplacX 'X' with 'X'"
 ```
 
-## 4. Reformat a log line
+## 4. Count characters
 
-Implement the `Reformat` function that reformats the log line, putting the message first and the log level after it in parentheses:
+After using this library, users are asking for the ability to count the number of characters in log lines.
```

Comment:
Story idea from real world experience: The system that should display the logs later has a limit on the number of characters per log entry it can display. For this reason you need a helper function that returns the number of characters. 

Even nicer would be when the function accepts a string and a character limit and returns a bool whether or not the limit was met. However you would need to fine tune the test cases to make sure the student used the correct way of counting.